pkg/ride/tree_estimatorV3.go

```diff
@@ -9,6 +9,7 @@ import (
 type fd struct {
 	cost   int
 	usages []string
+	dirty  bool
```

Comment:
Could you please add a comment what `dirty` means in this context or change the name of the variable